Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp

```diff
@@ -101,17 +109,16 @@ void GraphicsLayerCoordinated::setNeedsDisplayInRect(const FloatRect& initialRec
     if (rect.isEmpty())
         return;
 
-    auto& rects = m_dirtyRegion.rects;
-    bool alreadyRecorded = std::any_of(rects.begin(), rects.end(), [&](auto& dirtyRect) {
-        return dirtyRect.contains(rect);
-    });
-    if (alreadyRecorded)
-        return;
+    addRepaintRect(rect);
 
-    rects.append(rect);
-    noteLayerPropertyChanged(Change::DirtyRegion, ScheduleFlush::Yes);
+    if (!m_dirtyRegion) {
+        static constexpr FloatSize minDamageSize = { 512, 512 };
```

Comment:
Why the 512 hardcoded here? Do you want to use the default tile size here?

Reply:
the bigger the damage rect, the bigger it's resolution i.e. it can store more damage rects thus describing the changed region in more precise way